File: Algorithm/Other/at coder/XOR/A33.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "strconv"
    "strings"
)

func main() {
    reader := bufio.NewReader(os.Stdin)

    // 1行目: 山の数Nを取得
    line, _ := reader.ReadString('\n')
    N, _ := strconv.Atoi(strings.TrimSpace(line))

    // 2行目: 石の数の配列Aを取得
    line, _ = reader.ReadString('\n')
    tokens := strings.Fields(line)

    xorSum := 0
    for i := 0; i < N; i++ {
        a, _ := strconv.Atoi(tokens[i])
        xorSum ^= a
    }

    if xorSum == 0 {
        fmt.Println("Second")
    } else {
        fmt.Println("First")
    }
}

File: Algorithm/Other/at coder/XOR/A33.js
// 標準入力を受け取る
const fs = require("fs");

const input = fs.readFileSync("/dev/stdin", "utf8").trim().split(/\s+/);
const N = parseInt(input[0]);
const A = input.slice(1).map(Number);

let xorSum = 0;
for (let i = 0; i < N; i++) {
	xorSum ^= A[i];
}

console.log(xorSum === 0 ? "Second" : "First");
